grazer/src/main/java/edu/asu/diging/grazer/web/FileUploadController.java

```diff
@@ -0,0 +1,78 @@
+package edu.asu.diging.grazer.web;
+
+import java.io.IOException;
+import java.security.Principal;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import edu.asu.diging.grazer.web.fileUpload.FileUploadFormImpl;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesImpl;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesMetadataImpl;
+import edu.asu.diging.grazer.core.fileupload.service.impl.FileUploadServiceImpl;
+import edu.asu.diging.grazer.web.validator.FormValidator;
+
+@Controller
+public class FileUploadController {
+    
+    @Autowired
+    private FileUploadServiceImpl service;
+    
+    @Autowired
+    private FormValidator formValidator;
+    
+    @Autowired
+    private TransformationFilesImpl transformation;
+    
+    @Autowired
+    private TransformationFilesMetadataImpl metadata;
```

Comment:
those should not be autowired but created in the method where you need them. stateless variables should never be global (because then they are not stateless anymore).